sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StatePartitionReader.scala

```diff
@@ -68,10 +69,18 @@ abstract class StatePartitionReaderBase(
     stateVariableInfoOpt: Option[TransformWithStateVariableInfo],
     stateStoreColFamilySchemaOpt: Option[StateStoreColFamilySchema])
   extends PartitionReader[InternalRow] with Logging {
+  protected val schemaForValueRow: StructType =
```

Comment:
Nits: we can use `private` as this is not used in the inherited class.

Reply:
Also perhaps we can leave a small comment here that we are only using this for state variables dummy value state in `createAndInit()`.

Reply:
Done